.appveyor.yml

```diff
@@ -0,0 +1,283 @@
+# This CI setup provides a largely homogeneous configuration across all
+# major platforms (Windows, MacOS, and Linux). The aim of this test setup is
+# to create a "native" platform experience, using as few cross-platform
+# helper tools as possible.
+#
+# On Linux/Mac a virtualenv is used for testing. The effective virtual env
+# is available under ~/VENV.
+#
+# All workers support remote login. Login details are shown at the top of each
+# CI run log.
+#
+# - Linux/Mac workers (via SSH):
+#
+#   - A permitted SSH key must be defined in an APPVEYOR_SSH_KEY environment
+#     variable (via the appveyor project settings)
+#
+#   - SSH login info is given in the form of: 'appveyor@67.225.164.xx -p 22xxx'
+#
+#   - Login with:
+#
+#     ssh -o StrictHostKeyChecking=no <LOGIN>
+#
+#   - to prevent the CI run from exiting, `touch` a file named `BLOCK` in the
+#     user HOME directory (current directory directly after login). The session
+#     will run until the file is removed (or 60 min have passed)
+#
+# - Windows workers (via RDP):
+#
+#   - An RDP password should be defined in an APPVEYOR_RDP_PASSWORD environment
+#     variable (via the appveyor project settings), or a random password is used
+#     every time
+#
+#   - RDP login info is given in the form of IP:PORT
+#
+#   - Login with:
+#
+#     xfreerdp /cert:ignore /dynamic-resolution /u:appveyor /p:<PASSWORD> /v:<LOGIN>
+#
+#   - to prevent the CI run from exiting, create a textfile named `BLOCK` on the
+#     Desktop (a required .txt extension will be added automatically). The session
+#     will run until the file is removed (or 60 min have passed)
+
+
+# make repository clone cheap
+shallow_clone: true
+
+
+environment:
+  DATALAD_TESTS_SSH: 1
+
+  # Do not use `image` as a matrix dimension, to have fine-grained control over
+  # what tests run on which platform
+  # The ID variable had no impact, but sorts first in the CI run overview
+  # an intelligible name can help to locate a specific test run
+  matrix:
+    ## going for ~30min CI job duration to leave ample time for debugging and
+    ## get fast turn around via multi-job parallelization
+
+    # Windows core tests
+    - ID: WinP38core
+      # ~35 min
+      DTS: datalad.core
+      APPVEYOR_BUILD_WORKER_IMAGE: Visual Studio 2019
+      # Python version specification is non-standard on windows
+      PY: 38-x64
+    - ID: WinP38a1
+      # ~30min
+      DTS: >
+          datalad.cmdline
+          datalad.customremotes
+          datalad.distribution
```

Comment:
I would have really preferred if all matrix runs tested entire `datalad` package and not subselection of modules.  That is what we do in datalad/git-annex already and seems to be "almost there" for both windows (just few failing and/or unmarked as known fail/skip) and OSX (some odd issues relating to symlinks in the path AFAIK).  See https://github.com/datalad/git-annex/blob/master/.github/workflows/template/build-%7B%7Bostype%7D%7D.yaml.j2#L602 and actions results there for inspiration

- If splitting of runs is needed, best to split like how we do in .travis.yml separating out slow/integration tests into separate runs
  - that would 1) guarantee that we would not miss tests whenever new test files added; 2) needs no CI adjustment if any test file is removed/moved etc

Reply:
https://www.appveyor.com/pricing/ says

All plans have maximum build job execution time of 60 minutes. For longer build times contact us.

Since we are paying, I think it would be very reasonable to contact them and ask to extend e.g. to 180 min, which would also provide better/sufficient time for interactive sessions whenever needed etc.

> but only specific flavors somehow declared "inside"

what do you mean beyond tags which we already have (`slow`, `turtle`, etc)?

Reply:
I don't mind asking for longer times. But I very much prefer a complete test run in ~30min, hence the aim for 30min chunks and the commitment to pay 10 workers. If the pro is fast turn around and the con is only the risk of not running some test in a newly introduced module, than I'd say the risk is negligible.

Reply:
FWIW my 1c: I would prefer a complete run under a minute, and we could achieve it by running even less tests... but the point is to test well, and speed is IMHO a factor to minimize in code, not by tests sub-selection as we do regularly discover problems in niche tests, and delays in their discovery waste our time later to identify how/when they really emerged.
But oh well, as long as this PR does not drop already existing testing, at least it would not be worse than before, and it would be under 30 minutes making you happy.

Reply:
I am not sure how you got the impression that tests are dropped. The number of tests that do run has gone up substantially. In fact, never before was that ever even attempted for any windows test setup. It is standard now, bringing it much closer to the level of other OSes.

Reply:
> In fact, never before was that ever even attempted for any windows test setup

I am confused, since as I have mentioned we just run on entire `datalad` in datalad/git-annex.  BTW -- for the first time in history all marked to be not-skipped tests are passing now against maint/release/master! (git-annex itself has an issue in its tests, [already reported](https://git-annex.branchable.com/bugs/fresh_test_fails_for___34__trust__58____34___-_trust_failed_/)) . Hopefully git-annex itself gets a fix soon and we would turn green to establish a baseline there!

FTR, https://github.com/datalad/git-annex/actions/runs/473493901 

maint: Ran 1175 tests in 5706.491s OK (SKIP=318)
master: Ran 1210 tests in 4514.621s OK (SKIP=320)

Reply:
> I am not sure how you got the impression that tests are dropped. 

I never stated that something is dropped. We already did similar sub-selection before, I just expressed my opinion/desire to avoid such subselection, and that this PR is ok with me as long as no tests are dropped from before (for some reason I do not see coverage report now, so cannot attest one way or another)